tests/unit/dataclasses/test_template_generation.py

```diff
@@ -145,7 +145,7 @@ def test_csv_generation_file(
         template = template.inline_dependencies()
     bindings, filled = template.fill(BLDG, include_optional=include_optional)
 
-    with NamedTemporaryFile(mode="w", suffix=".csv") as dest:
+    with NamedTemporaryFile(mode="w", suffix=".csv", delete=False) as dest:
```

Comment:
Same thing here; make sure to call `dest.flush()` so that we are consistent across the tests